statics/workflows/ping-mesh.yaml

```diff
@@ -0,0 +1,102 @@
+---
+UUID: "784c3329-0f47-449b-5c58-2d207bcfb501"
+Name: "Ping Mesh (ICMP/TCP/UDP)"
+Description: "Check Connectivity from Multiple Source to Single Destination"
+Parameters:
+  - Name: protocol
+    Description: Protocol
+    Type:  choice
+    Default: icmp4
+    Values:
+      - Description: "Protocol : ICMPv4/Echo request"
+        Value: icmp4
+      - Description: "Protocol : TCP/IPv4"
+        Value: tcp4
+      - Description: "Protocol : UDP/IPv4"
+        Value: udp4
+  - Name: source
+    Description: Source Nodes (Enter a Gremlin Query to select Source Nodes)
+    Type: string
+  - Name: destination
+    Description: Destination Node (Select Destination Node)
+    Type: node
+Source: |
+  function PingMesh(protocol, src, dst) {
+    var sources = [];
+    var result = {};
+    var pktform = {};
+    client.gremlin.query(src).then(function(nodes) {
+      nodes.forEach(function(node) {
+        sources.push(node.Metadata.TID);
+      });
+    })
+    var capture = new Capture();
+    capture.GremlinQuery = "G.V().Has('TID', '" + dst + "')";
+    var packetInjection = new PacketInjection();
+    return client.captures.create(capture).then(function (c) {
+      capture = c
+    }).then(function () {
+      return sleep(1000)
+    }).then(function () {
+      sources.forEach(function(s) {
+        packetInjection.Src = "G.V().Has('TID', '" + s + "')"
```

Comment:
@dvandra packetInjection need to be created/used per source

Do a test with 3 nodes, launch it multiple times, look at the flows found, and you will see that you don't inject the right packets

Reply:
It is working now!